### PR TITLE
Update deps. and fix some eslint stuff

### PR DESCRIPTION
**Pull request by chrisnojima**
So my eslint got broken recently so i decided to fix that and also cleanup some accumulated eslint issues. Along the way i updated a bunch of deps, babel feels faster (they claim its like 3x faster)

@keybase/react-hackers

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,10 +1,15 @@
 {
     "parser": "babel-eslint",
     "extends": ["standard", "standard-react"],
+    "globals": {
+        "__DEV__": false
```
Comment (chrisnojima) [inline .eslintrc]: let's just have __DEV__ be a global so we don't need that comment to ignore it all the time

File: shared/reducers/signup.js [hunk 2]
```diff
@@ -33,6 +33,7 @@ const initialState: SignupState = {
   phase: 'inviteCode'
 }
 
+/* eslint-disable no-fallthrough */
```
Comment (chrisnojima) [inline shared/reducers/signup.js]: annoyingly eslint gets confused here and if we do add breaks then flow complains. i *think* eslint 2.0 fixes this but its not out of rc yet

File: .eslintrc [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+    },
     "plugins": [
         "flow-vars"
     ],
     "rules": {
+        "flow-vars/define-flow-type": 1,
+        "flow-vars/use-flow-type": 1,
```
Comment (MarcoPolo) [inline .eslintrc]: What do these rules do?

**Comment by MarcoPolo** [comment]
:+1:

File: .eslintrc (hunk 3 above)
Comment (chrisnojima) [inline .eslintrc]: it wans't needed before but is now for the plugin to work, it fixes missing reactElement and etc

**Pull request merged**